chainer/datasets/transform_dataset.py

```diff
@@ -0,0 +1,43 @@
+from chainer.dataset import dataset_mixin
+
+
+class TransformDataset(dataset_mixin.DatasetMixin):
+
+    """Dataset that indexes the base dataset and transforms the data.
+
+    This dataset wraps the base dataset by modifying the behavior of the base
+    dataset's :meth:`__getitem__`. Arrays returned by :meth:`__getitem__` of
+    the base dataset with integer as an argument are transformed by the given
+    function :obj:`transform`.
+
+    The function :obj:`transform` takes, as an argument, :obj:`in_data`, which
+    is the output of the base dataset's :meth:`__getitem__`, and returns
+    the transformed arrays as output. Please see the following example.
+
+    >>> from chainer.datasets import get_mnist
+    >>> from chainer.datasets import TransformDataset
+    >>> dataset, _ = get_mnist()
+    >>> def transform(in_data):
+    >>>     img, label = in_data
```

Comment:
When the line is continued without evaluation by the interpreter, you need to begin the line with `...` instead of `>>>` here.